src/util/system.cpp

```diff
@@ -402,6 +402,27 @@ void ArgsManager::WarnForSectionOnlyArgs()
     }
 }
 
+// Section names to be recognized in the config file.
+static const std::set<std::string> G_CONFIG_SECTIONS{
+    CBaseChainParams::REGTEST,
+    CBaseChainParams::TESTNET,
+    CBaseChainParams::MAIN
+};
+
+void ArgsManager::WarnForUnrecognizedSection() const
+{
+    std::set<std::string> diff;
+
+    LOCK(cs_args);
+    std::set_difference(
+        m_config_sections.begin(), m_config_sections.end(),
+        G_CONFIG_SECTIONS.begin(), G_CONFIG_SECTIONS.end(),
+        std::inserter(diff, diff.end()));
+    for (const auto& name : diff) {
+        LogPrintf("Warning: Section [%s] is not recognized.\n", name);
```

Comment:
Hmm. This only ends up in the debug log. However, I think that warnings could as well be printed out. (See how `InitError()` does it.)

Not sure if that suggestion makes sense, but if applied you could probably also write a test case like this:

```diff
diff --git a/test/functional/feature_config_args.py b/test/functional/feature_config_args.py
index 88a9aadc7b..55ef46d253 100755
--- a/test/functional/feature_config_args.py
+++ b/test/functional/feature_config_args.py
@@ -33,6 +33,11 @@ class ConfArgsTest(BitcoinTestFramework):
             conf.write('server=1\nrpcuser=someuser\nrpcpassword=some#pass')
         self.nodes[0].assert_start_raises_init_error(expected_msg='Error reading configuration file: parse error on line 3, using # in rpcpassword can be ambiguous and should be avoided')
 
+        with open(inc_conf_file_path, 'w', encoding='utf-8') as conf:
+            conf.write('testneettt.datadir=1')
+        self.restart_node(0)
+        self.nodes[0].stop_node(expected_stderr='Warning: Section [testneettt] is not recognized.')
+
         with open(inc_conf_file_path, 'w', encoding='utf-8') as conf:
             conf.write('')  # clear

Reply:
@MarcoFalke Thanks for your suggestion. I think it will be more useful. I'll try it.

Reply:
done.

Reply:
I could not write a test case for WarnForSectionOnlyArgs().
Because I could not found simple way that how to produce the situation that WarnForSectionOnlyArgs() warn and to continue to run through the test.